### PR TITLE
[Self chat] Return logs

### PR DESCRIPTION
**Pull request by emilydinan**
**Patch description**
Edit the self chat function to return the logs.

### CODE REVIEW

File: parlai/scripts/self_chat.py [hunk 1]
```diff
@@ -72,6 +72,14 @@ def self_chat(opt, print_parser=None):
     if hasattr(agent2, 'id'):
         agent2.id = agent2.id + "2"
 
+    # Check for `selfchat` in the task name
+    if 'selfchat' not in opt['task']:
+        warn_once(
+            'You are using self chat with task {}.'.format(opt['task']) +
+            'If your task has an existing self chat world, then run with '
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: You missed a space. Also you can use `f'strings'` if you want!

File: parlai/scripts/self_chat.py [hunk 2]
```diff
@@ -103,7 +111,9 @@ def self_chat(opt, print_parser=None):
     if opt.get('display_examples'):
         print('-- end of episode --')
 
+    logger.reset_world()  # add to logs
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: this is so confusing, how does this add to logs?!

File: parlai/scripts/self_chat.py (hunk 1 above)
Comment (stephenroller) [reply parlai/scripts/self_chat.py]: I don't suppose we could check if such a self chat world exists but such is.

File: parlai/scripts/self_chat.py (hunk 2 above)
Comment (jaseweston) [reply parlai/scripts/self_chat.py]: hmm yea not sure what this is.. was it not working before?.. oh like the last episode maybe wasn't flushed, or..?

Comment (emilydinan) [reply parlai/scripts/self_chat.py]: yeah it only flushes when the episode is done

**Pull request merged**